task/control-plane-agent/src/update/rot.rs

```diff
@@ -72,15 +72,22 @@ impl ComponentUpdater for RotUpdate {
         }
         // Can we lock the update buffer?
         let buffer =
-            buffer.borrow(SpComponent::ROT, Self::BLOCK_SIZE).map_err(
+            buffer.borrow(update.component, Self::BLOCK_SIZE).map_err(
                 |component| SpError::OtherComponentUpdateInProgress(component),
             )?;
 
         // Which target are we updating?
-        let target = match update.slot {
-            0 => UpdateTarget::ImageA,
-            1 => UpdateTarget::ImageB,
-            _ => return Err(SpError::InvalidSlotForComponent),
+        let target = match update.component {
+            SpComponent::ROT => match update.slot {
+                0 => UpdateTarget::ImageA,
+                1 => UpdateTarget::ImageB,
+                _ => return Err(SpError::InvalidSlotForComponent),
+            },
+            SpComponent::STAGE0 => match update.slot {
+                0 => UpdateTarget::Bootloader,
+                _ => return Err(SpError::InvalidSlotForComponent),
+            },
+            _ => unreachable!(),
         };
```

Comment:
This might be a little cleaner if we match on the tuple instead of nesting matches:

```rust
        let target = match (update.component, update.slot) {
            (SpComponent::ROT, 0) => UpdateTarget::ImageA,
            (SpComponent::ROT, 1) => UpdateTarget::ImageB,
            (SpComponent::STAGE0, 0) => UpdateTarget::Bootloader,
            _ => return Err(SpError::InvalidSlotForComponent),
        };
```

This discards the unreachable case of "someone called us with a component other than ROT/STAGE0" and would return `InvalidSlotForComponent`. We could add a

```rust
(SpComponent::STAGE0 | SpComponent::ROT, _) => unreachable!()`
```

case if we think that's important? I could go either way on that.